src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetter.java

```diff
@@ -70,34 +75,70 @@ public AbstractBigQueryActionConfig getConfig() {
   }
 
   @Override
-  public void run(ActionContext context) throws Exception {
+  public void run(ActionContext context) {
     config.validate(context.getFailureCollector());
 
     QueryJobConfiguration queryConfig = config.getQueryJobConfiguration(context.getFailureCollector());
     JobId jobId = JobId.newBuilder().setRandomJob().build();
 
     // API request - starts the query.
-    Credentials credentials = config.getServiceAccount() == null ?
-      null : GCPUtils.loadServiceAccountCredentials(config.getServiceAccount(),
-                                                    config.isServiceAccountFilePath());
+    Credentials credentials = null;
+    try {
+      credentials = config.getServiceAccount() == null ? null :
+        GCPUtils.loadServiceAccountCredentials(config.getServiceAccount(), config.isServiceAccountFilePath());
+    } catch (Exception e) {
+      context.getFailureCollector().addFailure(
+          String.format("Failed to load service account credentials, %s: %s",
+              e.getClass().getName(), e.getMessage()), null).withStacktrace(e.getStackTrace());
+      context.getFailureCollector().getOrThrowException();
+    }
     BigQuery bigQuery = GCPUtils.getBigQuery(config.getProject(), credentials, null);
     Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
 
     LOG.info("Executing SQL as job {}.", jobId.getJob());
     LOG.debug("The BigQuery SQL  {}", queryConfig.getQuery());
 
     // Wait for the query to complete
-    queryJob.waitFor();
+    try {
+      queryJob.waitFor();
+    } catch (BigQueryException | InterruptedException e) {
+      String errorMessage = String.format("The query job was interrupted, %s: %s",
+          e.getClass().getName(), e.getMessage());
+      if (e instanceof BigQueryException) {
+        throw BigQueryErrorUtil.getProgramFailureException(errorMessage,
+            ((BigQueryException) e).getReason(), e);
+      }
+      throw ErrorUtils.getProgramFailureException(
+          new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), errorMessage, errorMessage,
+          ErrorType.UNKNOWN, true, e);
+    }
 
     // Check for errors
     if (queryJob.getStatus().getError() != null) {
-      throw new RuntimeException(queryJob.getStatus().getExecutionErrors().toString());
+      String error = queryJob.getStatus().getExecutionErrors().toString();
+      ErrorType type = BigQueryErrorUtil.getErrorType(queryJob.getStatus().getError().getReason());
+      throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+          error, error, type, true, null);
```

Comment:
`errorReason = String.format("The bigquery job failed with reason: %s", queryJob.getStatus.getError.getReason())`